source/agora/node/Ledger.d

```diff
@@ -507,6 +507,31 @@ public class Ledger
     {
         return this.pool.hasTransactionHash(tx);
     }
+
+    /***************************************************************************
+
+        Generate the random seed reduced from the preimages for the provided
+        block height.
+
+        Params:
+            height = the desired block height to look up the images for
+
+        Returns:
+            the random seed
+
+    ***************************************************************************/
+
+    public Hash getValidatorRandomSeed (Height height) nothrow
```

Comment:
This is awkward code, it does not belong here.

There is nothing this function uses in the Ledger itself other than the enrollment manager. 

Why not move `this.enroll_man.getEnrolledUTXOs(keys)` inside of `EnrollmentManager.getRandomSeed` and remove the `in Hash[] keys` parameter. And then also provide the `EnrollmentManager` instance to the `Nominator` and make it call `getRandomSeed(slot_idx - 1)`?

Reply:
All right. I have already felt something weird about that and you have suggested the right approach. I'll apply it to the code. Thanks.